sdk/core/src/tasks/enrichments/identity/cpe/sync.rs

```diff
@@ -0,0 +1,147 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use platform::persistence::mongodb::{Service, Store};
+
+use crate::entities::datasets::{Purl2Cpes, PurlPlusId};
+use crate::entities::packages::Package;
+use crate::entities::tasks::Task;
+use crate::services::analytics::sboms::service::AnalyticService;
+use crate::services::purl2cpe::service::Purl2CpeService;
+use crate::tasks::TaskProvider;
+use crate::Error;
+
+/// Synchronizes the subset of [Package] entries with the kind of
+/// `dependency` and no `cpe` with a cpe derived from their purl(s).
+#[derive(Debug)]
+pub struct SyncTask {
+    store: Arc<Store>,
+    analytic_service: AnalyticService,
+    purl_2_cpe_service: Purl2CpeService,
+}
+
+#[async_trait]
+impl TaskProvider for SyncTask {
+    async fn run(&self, task: &mut Task) -> Result<HashMap<String, String>, Error> {
+        println!("==> fetching dependency Packages with no cpe");
+        let mut errors = HashMap::new();
+
+        let ids_and_purls_with_null_cpe: Vec<PurlPlusId> = self
+            .analytic_service
+            .get_dependant_package_purls_with_null_cpe()
+            .await?;
+
```

Comment:
Would be good to set the `task.count` here and then update the `task` document before running the iteration. We have an open bug for that in numerous other places.

Reply:
Done

Reply:
Almost.  We have to now update the task in document db after setting the count value so that we have a count in the db even if this completely fails at the end. That way we at least know the count we started with. I made this mistake in literally every task I wrote.

Reply:
Ha!  No problem

Reply:
Done